app/src/server/tasks/chargeInvoices.task.ts

```diff
@@ -0,0 +1,100 @@
+import defineTask from "./defineTask";
+import { prisma } from "~/server/db";
+import { error, success } from "~/utils/errorHandling/standardResponses";
+import {
+  createStripePaymentIntent,
+  getDefaultPaymentMethodId,
+  getPaymentMethods,
+  usdToCents,
+} from "../utils/stripe";
+import { env } from "~/env.mjs";
+import { emailAdminsAboutPaymentFailure } from "../utils/emails";
+
+export const chargeInvoices = defineTask({
+  id: "chargeInvoices",
+  handler: async (task) => {
+    const invoices = await prisma.invoice.findMany({
+      where: {
+        status: "PENDING",
+      },
+    });
+
+    for (const invoice of invoices) {
+      await chargeInvoice(invoice.id);
+    }
+  },
+  specDefaults: {
+    priority: 5,
+  },
+});
+
+export async function chargeInvoice(invoiceId: string) {
+  const invoice = await prisma.invoice.findUnique({
+    where: {
+      id: invoiceId,
+    },
+  });
+
+  if (invoice?.status !== "PENDING" || Number(invoice.amount) < 1) {
```

Comment:
If the invoice status is ERROR (due to an earlier payment method being declined) shouldn't we let the user pay in this function?

Reply:
I'd also prefer not to check the amount of the invoice here. If the invoice is below a certain threshold, we should just mark it as cancelled when we create it.

Reply:
Currently, an invoice would have a "PENDING" status if payment was declined. This way, the user can update payment details and pay it manually, or the system will charge automatically once the corn job executes.

When creating an invoice, I cancel it by default if the amount is <= 1. We can probably remove this check, but since the payment logic is different from the invoice creation logic, I explicitly added it.